File: src/ProjectR.Scripting/content/scripts/spells/BeastlySmash.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Beastly Smash"
end

function GetDescription()
    return "Smash your target. Simple and easy. Relatively strong."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .30
end

function GetDelay()
  return .18
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 3.5 * aAD - dDEF
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/BeastlySmash.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class BeastlySmash : SpellScriptBase
    {
        public override string Name { get { return "Beastly Smash"; } }
        public override string Description { get { return "Smash your target. Simple and easy. Relatively strong."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 30; } }
        public override double Delay { get { return .18; } }

        public override void SpellEffect(ICharacter caster, ICharacter target, double decayMod = 1)
        {
            var damage = 3.5 * AD(caster) - DEF(target);
            DealDamage(target, damage);
        }
    }
}